### ISSUE TITLE
Review and update documentation

### ISSUE THREAD

**Issue by wtrocki** [aerogear/charmil#216]
We should update/remove things like factory.

**Comment by jackdelahunt** [comment]
This is a big one for me atleast, the docs seem to be falling behind the project. This also makes it difficult as a contributor if you are trying to learn how a new feature works and there is no documentation for it. 

One of the examples for me is `starter` no sure if there is docs but I couldn't find them but if so please link me as I need a read up else we should try get that up and running.

**Comment by wtrocki** [comment]
Yes. I feel that this is what guys are working right now by building those examples and documenting experience.

@ankithans is that the case?

**Comment by ankithans** [comment]
yes @wtrocki, want to discuss again - Shall I grab a service from rhoas in charmil-host-example and one service in charmil-plugin-example? Or should I create some basic examples to show host interacting with plugin.

**Comment by wtrocki** [comment]
We discussed that and I went thru benefits of both approaches (using rhoas will get you instant code but some challenges), doing your own will require more time. My preference will be to use rhoas as example so host CLI can be RHOAS source code that is pushed to the host... and then plugin can be number of commands taken out of the host CLI.

Charmil is already in the repo so this will reduce amount of work you have

**Comment by wtrocki** [comment]
@ankithans let me know if you need quick sync

**Comment by ankithans** [comment]
clear & sorted.

charmil-host-example : rhoas
charmil-plugin-example : apicurio

**Comment by wtrocki** [comment]
With apicurio being - `rhoas service-registry artifacts`
